File: Leet Code Solutions/343. Integer Break/PralinKhaira-IntegerBreak.java
/*
Problem Link: https://leetcode.com/problems/integer-break/description/
*/

class Solution {
    public int integerBreak(int n) {
        if (n <= 3) return n - 1;
        int quotient = n / 3, remainder = n % 3;
        return remainder == 0 ? (int)Math.pow(3, quotient) : (remainder == 1 ? (int)Math.pow(3, quotient - 1) * 4 : (int)Math.pow(3, quotient) * 2);
    }
}

File: Leet Code Solutions/343. Integer Break/PralinKhaira-IntegerBreak.py
# Problem Link: https://leetcode.com/problems/integer-break/description/

class Solution:
    def integerBreak(self, n: int) -> int:
        return (n - 1) if n <= 3 else (3 ** (n // 3)) if n % 3 == 0 else (3 ** (n // 3 - 1) * 4) if n % 3 == 1 else (3 ** (n // 3) * 2)
